Modules/Survey/classes/class.ilSurveyImporter.php

```diff
@@ -85,8 +85,10 @@ function importXmlRepresentation($a_entity, $a_id, $a_xml, $a_mapping)
 				$GLOBALS['ilLog']->write(__METHOD__ . ': Cannot find xml definition: ' . $xml_file);
 				return false;
 			}
+			$GLOBALS['ilLog']->write("getQuestionPoolID = ".$this->getImport()->getConfig("Modules/Survey")->getQuestionPoolID());
+
+			$import = new SurveyImportParser($this->getImport()->getConfig("Modules/Survey")->getQuestionPoolID(), $xml_file, true, $a_mapping);
```

Comment:
I think this "-1" is ensured because it's forced if we don't pass the survey pool id.
The code:
class.ilObjSurveyGUI.php:
```
include_once("./Modules/Survey/classes/class.ilObjSurvey.php");
$newObj = new ilObjSurvey();
....
$error = $newObj->importObject($_FILES["importfile"], $form->getInput("spl"));
```

class.ilObjSurvey.php:
```
function importObject($file_info, $svy_qpl_id)
{
	if ($svy_qpl_id < 1) $svy_qpl_id = -1; 
```
class.SurveyImportParser.php:
```
if($this->spl_id < 0)
{
	$q_obj_id = $this->survey->getId();
}
```
						

The log:

> ilObjSurvey::importObject:1436 question pool id = -1
> ilSurveyImporter::importXmlRepresentation:357 getQuestionPoolID = -1
> SurveyImportParser::handlerBeginTag: spl_id =-1
> ilObjSurvey::importObject:1436 config(Modules/survey)->getQuestionPoolId =-1  

We can add this conditional:

```
if($this->getImport()->getConfig("Modules/Survey")->getQuestionPoolID() > 0)
{
	$import = new SurveyImportParser($this->getImport()->getConfig("Modules/Survey")->getQuestionPoolID(), $xml_file, true, $a_mapping);
}
else
{
	$import = new SurveyImportParser("-1", $xml_file, true, $a_mapping);
}
```